In [ ]:
import pandas as pd

# CSV 파일 경로 설정 (파일명은 실제 파일이 위치한 경로에 맞게 변경해주세요)
csv_file_path = 'new_jeju.csv'

# CSV 파일 읽어오기
df = pd.read_csv(csv_file_path,encoding='cp949')

# DataFrame 출력
df

콘텐츠아이디 콘텐츠분류                 제목  \
0     CNTS_000000000021903    숙박               해피펜션   
1     CONT_000000000501391   음식점             탐라 뚝배기   
2     CONT_000000000501399   음식점                토담골   
3     CNTS_000000000018301   음식점               우유부단   
4     CNTS_000000000020277   관광지                공천포   
...                    ...   ...                ...   
6794  CNTS_200000000015151   관광지        제주패들보드 서핑클럽   
6795  CNTS_200000000015157   관광지          카카오패밀리 세화   
6796  CNTS_200000000015164   관광지          강정 어촌체험마을   
6797  CNTS_200000000015266   음식점            미베이커리카페   
6798  CNTS_200000000015306  테마여행  제주마을산책 <서귀포 로컬여행>   

                           지번주소                       도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1          제주특별자치도 제주시 월대1길 4  33.494110   
1      제주특별자치도 서귀포시 표선면 표선리 879  제주특별자치도 서귀포시 표선면 표선백사로 127  33.326770   
2         제주특별자치도 제주시 아라이동 2425          제주특별자치도 제주시 간월동로 5  33.485904   
3     제주특별자치도 제주시 한림읍 금악리 142-2     제주특별자치도 제주시 한림읍 금악동길 38  33.347640   
4          제주특별자치도 서귀포시 남원읍 신례리        제주특별자치도 서귀포시 남원읍 신례리  33.267690   
...                         ...                         ...        ...   
6794                        NaN                         NaN        NaN   
6795                        NaN                         NaN        NaN   
6796                        NaN                         NaN        NaN   
6797       제주특별자치도 서귀포시 강정동 847       제주특별자치도 서귀포시 이어도로 780  33.236092   
6798                        NaN                         NaN        NaN   

              경도   평점    조회수  좋아요수  리뷰수  북마크수  소셜네트워크공유수     인기점수  사진공유수  
0     126.433780  NaN    979     0    0     1          0   1958.0    NaN  
1     126.836690  NaN   3120     0    0    12          1   6240.0    NaN  
2     126.543496  NaN   1973     0    0     4          0   3946.0    NaN  
3     126.328260  5.0  21460    14   38   253          5  43152.0   51.0  
4     126.645996  4.0   4779     0    9    26          2   9603.0   18.0  
...          ...  ...    ...   ...  ...   ...        ...      ...    ...  
6794         NaN  0.0      0     0    0     0          0      NaN    0.0  
6795         NaN  0.0      0     0    0     0          0      NaN    0.0  
6796         NaN  0.0      0     0    0     0          0      NaN    0.0  
6797  126.498936  0.0     18     1    0     0          0     39.0    0.0  
6798         NaN  0.0     41     0    0     0          0     82.0    0.0  

[6799 rows x 15 columns]

In [ ]:
df.describe()

위도           경도           평점            조회수         좋아요수  \
count  5348.000000  5348.000000  4889.000000    6799.000000  6799.000000   
mean     32.516425   123.132377     1.802209    5724.940432     1.425210   
std       5.411341    20.485514     2.279639   21821.600531     8.030408   
min       0.000000     0.000000     0.000000       0.000000     0.000000   
25%      33.286581   126.362526     0.000000     490.500000     0.000000   
50%      33.444030   126.514997     0.000000    1767.000000     0.000000   
75%      33.497777   126.668846     5.000000    4106.500000     1.000000   
max      37.568489   126.981613     5.000000  738269.000000   328.000000   

               리뷰수         북마크수    소셜네트워크공유수          인기점수        사진공유수  
count  6799.000000  6799.000000  6799.000000  5.915000e+03  4607.000000  
mean      3.986322    27.027357     4.110752  1.318873e+04     7.334708  
std      40.722364   153.551438    27.783960  4.666863e+04    47.381412  
min       0.000000     0.000000     0.000000  2.000000e+00     0.000000  
25%       0.000000     0.000000     0.000000  2.058000e+03     0.000000  
50%       0.000000     1.000000     0.000000  4.282000e+03     0.000000  
75%       1.000000     7.000000     2.000000  9.615500e+03     2.000000  
max    2555.000000  4017.000000  1213.000000  1.479728e+06  1263.000000

### 변수 각각 전처리 df->df_p

평점

In [ ]:
df['평점'].isna().sum()

1910

In [ ]:
df['평점'] = df['평점'].fillna(0)

In [ ]:
df['평점'].isna().sum()

0

지번주소 null값 지우기

In [ ]:
df_p = df.copy()

In [ ]:
df_p['지번주소'].isnull().sum()

1364

In [ ]:
df_p['지번주소'].isnull()

0       False
1       False
2       False
3       False
4       False
        ...  
6794     True
6795     True
6796     True
6797    False
6798     True
Name: 지번주소, Length: 6799, dtype: bool

In [ ]:
df_p = df_p[df_p['지번주소'].isnull()==False]

In [ ]:
6798-5435

1363

In [ ]:
df_p

콘텐츠아이디 콘텐츠분류           제목                       지번주소  \
0     CNTS_000000000021903    숙박         해피펜션     제주특별자치도 제주시 외도2동 317-1   
1     CONT_000000000501391   음식점       탐라 뚝배기   제주특별자치도 서귀포시 표선면 표선리 879   
2     CONT_000000000501399   음식점          토담골      제주특별자치도 제주시 아라이동 2425   
3     CNTS_000000000018301   음식점         우유부단  제주특별자치도 제주시 한림읍 금악리 142-2   
4     CNTS_000000000020277   관광지          공천포       제주특별자치도 서귀포시 남원읍 신례리   
...                    ...   ...          ...                        ...   
6705  CNTS_200000000015227   관광지        한형수정원   제주특별자치도 제주시 한림읍 협재리 1352   
6759  CNTS_200000000015249   음식점  한형수정원 테쉬폰카페   제주특별자치도 제주시 한림읍 협재리 1352   
6760  CNTS_200000000014966    쇼핑          유화당     제주특별자치도 서귀포시 서귀동 481-1   
6763  CNTS_200000000015247   음식점    트레비어 제주본점  제주특별자치도 제주시 삼도일동 561-2 1층   
6797  CNTS_200000000015266   음식점      미베이커리카페       제주특별자치도 서귀포시 강정동 847   

                              도로명주소         위도          경도   평점    조회수  좋아요수  \
0                제주특별자치도 제주시 월대1길 4  33.494110  126.433780  0.0    979     0   
1        제주특별자치도 서귀포시 표선면 표선백사로 127  33.326770  126.836690  0.0   3120     0   
2                제주특별자치도 제주시 간월동로 5  33.485904  126.543496  0.0   1973     0   
3           제주특별자치도 제주시 한림읍 금악동길 38  33.347640  126.328260  5.0  21460    14   
4              제주특별자치도 서귀포시 남원읍 신례리  33.267690  126.645996  4.0   4779     0   
...                             ...        ...         ...  ...    ...   ...   
6705        제주특별자치도 제주시 한림읍 옹포2길 44  33.397039  126.250255  0.0     92     0   
6759        제주특별자치도 제주시 한림읍 옹포2길 44  33.397039  126.250255  0.0     42     0   
6760         제주특별자치도 서귀포시 태평로 409-2  33.245194  126.563893  0.0      0     0   
6763  제주특별자치도 제주시 서광로 192 (삼도일동) 1층        NaN         NaN  0.0      0     0   
6797          제주특별자치도 서귀포시 이어도로 780  33.236092  126.498936  0.0     18     1   

      리뷰수  북마크수  소셜네트워크공유수     인기점수  사진공유수  
0       0     1          0   1958.0    NaN  
1       0    12          1   6240.0    NaN  
2       0     4          0   3946.0    NaN  
3      38   253          5  43152.0   51.0  
4       9    26          2   9603.0   18.0  
...   ...   ...        ...      ...    ...  
6705    0     0          0    184.0    0.0  
6759    0     0          0     84.0    0.0  
6760    0     0          0      NaN    0.0  
6763    0     0          0      NaN    0.0  
6797    0     0          0     39.0    0.0  

[5435 rows x 15 columns]

위도

In [ ]:
df_p['위도'].isnull().sum()

228

In [ ]:
df_p = df_p[df_p['위도'].isnull()==False]

In [ ]:
df_p['위도'].isnull().sum()

0

경도

In [ ]:
df_p['경도'].isnull().sum()

0

In [ ]:
df_p

콘텐츠아이디  콘텐츠분류                         제목  \
0     CNTS_000000000021903     숙박                       해피펜션   
1     CONT_000000000501391    음식점                     탐라 뚝배기   
2     CONT_000000000501399    음식점                        토담골   
3     CNTS_000000000018301    음식점                       우유부단   
4     CNTS_000000000020277    관광지                        공천포   
...                    ...    ...                        ...   
6529  CNTS_200000000015288  축제/행사  JSW SPRING BREEZE CONCERT   
6705  CNTS_200000000015227    관광지                      한형수정원   
6759  CNTS_200000000015249    음식점                한형수정원 테쉬폰카페   
6760  CNTS_200000000014966     쇼핑                        유화당   
6797  CNTS_200000000015266    음식점                    미베이커리카페   

                              지번주소                           도로명주소         위도  \
0           제주특별자치도 제주시 외도2동 317-1              제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 서귀포시 표선면 표선리 879      제주특별자치도 서귀포시 표선면 표선백사로 127  33.326770   
2            제주특별자치도 제주시 아라이동 2425              제주특별자치도 제주시 간월동로 5  33.485904   
3        제주특별자치도 제주시 한림읍 금악리 142-2         제주특별자치도 제주시 한림읍 금악동길 38  33.347640   
4             제주특별자치도 서귀포시 남원읍 신례리            제주특별자치도 서귀포시 남원읍 신례리  33.267690   
...                            ...                             ...        ...   
6529  제주특별자치도 서귀포시 안덕면 서광리 산 35-16  제주특별자치도 서귀포시 안덕면 신화역사로304번길 98  33.294647   
6705      제주특별자치도 제주시 한림읍 협재리 1352         제주특별자치도 제주시 한림읍 옹포2길 44  33.397039   
6759      제주특별자치도 제주시 한림읍 협재리 1352         제주특별자치도 제주시 한림읍 옹포2길 44  33.397039   
6760        제주특별자치도 서귀포시 서귀동 481-1          제주특별자치도 서귀포시 태평로 409-2  33.245194   
6797          제주특별자치도 서귀포시 강정동 847           제주특별자치도 서귀포시 이어도로 780  33.236092   

              경도   평점    조회수  좋아요수  리뷰수  북마크수  소셜네트워크공유수     인기점수  사진공유수  
0     126.433780  0.0    979     0    0     1          0   1958.0    NaN  
1     126.836690  0.0   3120     0    0    12          1   6240.0    NaN  
2     126.543496  0.0   1973     0    0     4          0   3946.0    NaN  
3     126.328260  5.0  21460    14   38   253          5  43152.0   51.0  
4     126.645996  4.0   4779     0    9    26          2   9603.0   18.0  
...          ...  ...    ...   ...  ...   ...        ...      ...    ...  
6529  126.314558  0.0    115     0    0     0          0    230.0    0.0  
6705  126.250255  0.0     92     0    0     0          0    184.0    0.0  
6759  126.250255  0.0     42     0    0     0          0     84.0    0.0  
6760  126.563893  0.0      0     0    0     0          0      NaN    0.0  
6797  126.498936  0.0     18     1    0     0          0     39.0    0.0  

[5207 rows x 15 columns]

In [ ]:
df_p['제목'].isna().sum()

0

In [ ]:
df_p['콘텐츠분류'].isna().sum()

0

In [ ]:
df_p['조회수'].isna().sum()

0

In [ ]:
df_p['좋아요수'].isna().sum()

0

In [ ]:
df_p['리뷰수'].isna().sum()

0

In [ ]:
df_p['북마크수'].isna().sum()

0

In [ ]:
df_p['소셜네트워크공유수'].isna().sum()

0

null값을 0으로 바꿔줌

In [ ]:
df_p['인기점수'].isna().sum()

46

In [ ]:
df_p['인기점수'] = df_p['인기점수'].fillna(0)

<ipython-input-137-e96e106e1471>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_p['인기점수'] = df_p['인기점수'].fillna(0)


In [ ]:
df_p['인기점수'].isna().sum()

0

In [ ]:
df_p['사진공유수'].isna().sum()

2065

In [ ]:
df_p['사진공유수'] = df_p['사진공유수'].fillna(0)

<ipython-input-139-0f7c400adae1>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_p['사진공유수'] = df_p['사진공유수'].fillna(0)


In [ ]:
df_p['사진공유수'].isna().sum()

0

In [ ]:
df_p.shape

(5207, 15)

평점 + 사진공유수 널값 너무 많음. 빼는게 좋음


# 동까지 뽑아내기 -> df_6 완



In [ ]:
df_6 = df_p.copy()

In [ ]:
df_6['지번주소']

0             제주특별자치도 제주시 외도2동 317-1
1           제주특별자치도 서귀포시 표선면 표선리 879
2              제주특별자치도 제주시 아라이동 2425
3          제주특별자치도 제주시 한림읍 금악리 142-2
4               제주특별자치도 서귀포시 남원읍 신례리
                    ...             
6529    제주특별자치도 서귀포시 안덕면 서광리 산 35-16
6705        제주특별자치도 제주시 한림읍 협재리 1352
6759        제주특별자치도 제주시 한림읍 협재리 1352
6760          제주특별자치도 서귀포시 서귀동 481-1
6797            제주특별자치도 서귀포시 강정동 847
Name: 지번주소, Length: 5207, dtype: object

In [ ]:
df_6['지번주소'].str.split().str[0]

0       제주특별자치도
1       제주특별자치도
2       제주특별자치도
3       제주특별자치도
4       제주특별자치도
         ...   
6529    제주특별자치도
6705    제주특별자치도
6759    제주특별자치도
6760    제주특별자치도
6797    제주특별자치도
Name: 지번주소, Length: 5207, dtype: object

    5207 -> 4988
    제주 아닌것들 뺌 218개 <-추후에 더 정제하는게 좋을거같음 일단 바쁘니까 패스

In [ ]:
# '지번주소' 열에서 '제주특별자치도'이 들어가는 행 추출

df_6['도'] = df_6['지번주소'].str.split().str[0]
df_6 = df_6[df_6['도'] == '제주특별자치도']
df_6

콘텐츠아이디  콘텐츠분류                         제목  \
0     CNTS_000000000021903     숙박                       해피펜션   
1     CONT_000000000501391    음식점                     탐라 뚝배기   
2     CONT_000000000501399    음식점                        토담골   
3     CNTS_000000000018301    음식점                       우유부단   
4     CNTS_000000000020277    관광지                        공천포   
...                    ...    ...                        ...   
6529  CNTS_200000000015288  축제/행사  JSW SPRING BREEZE CONCERT   
6705  CNTS_200000000015227    관광지                      한형수정원   
6759  CNTS_200000000015249    음식점                한형수정원 테쉬폰카페   
6760  CNTS_200000000014966     쇼핑                        유화당   
6797  CNTS_200000000015266    음식점                    미베이커리카페   

                              지번주소                           도로명주소         위도  \
0           제주특별자치도 제주시 외도2동 317-1              제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 서귀포시 표선면 표선리 879      제주특별자치도 서귀포시 표선면 표선백사로 127  33.326770   
2            제주특별자치도 제주시 아라이동 2425              제주특별자치도 제주시 간월동로 5  33.485904   
3        제주특별자치도 제주시 한림읍 금악리 142-2         제주특별자치도 제주시 한림읍 금악동길 38  33.347640   
4             제주특별자치도 서귀포시 남원읍 신례리            제주특별자치도 서귀포시 남원읍 신례리  33.267690   
...                            ...                             ...        ...   
6529  제주특별자치도 서귀포시 안덕면 서광리 산 35-16  제주특별자치도 서귀포시 안덕면 신화역사로304번길 98  33.294647   
6705      제주특별자치도 제주시 한림읍 협재리 1352         제주특별자치도 제주시 한림읍 옹포2길 44  33.397039   
6759      제주특별자치도 제주시 한림읍 협재리 1352         제주특별자치도 제주시 한림읍 옹포2길 44  33.397039   
6760        제주특별자치도 서귀포시 서귀동 481-1          제주특별자치도 서귀포시 태평로 409-2  33.245194   
6797          제주특별자치도 서귀포시 강정동 847           제주특별자치도 서귀포시 이어도로 780  33.236092   

              경도   평점    조회수  좋아요수  리뷰수  북마크수  소셜네트워크공유수     인기점수  사진공유수  \
0     126.433780  0.0    979     0    0     1          0   1958.0    0.0   
1     126.836690  0.0   3120     0    0    12          1   6240.0    0.0   
2     126.543496  0.0   1973     0    0     4          0   3946.0    0.0   
3     126.328260  5.0  21460    14   38   253          5  43152.0   51.0   
4     126.645996  4.0   4779     0    9    26          2   9603.0   18.0   
...          ...  ...    ...   ...  ...   ...        ...      ...    ...   
6529  126.314558  0.0    115     0    0     0          0    230.0    0.0   
6705  126.250255  0.0     92     0    0     0          0    184.0    0.0   
6759  126.250255  0.0     42     0    0     0          0     84.0    0.0   
6760  126.563893  0.0      0     0    0     0          0      0.0    0.0   
6797  126.498936  0.0     18     1    0     0          0     39.0    0.0   

            도  
0     제주특별자치도  
1     제주특별자치도  
2     제주특별자치도  
3     제주특별자치도  
4     제주특별자치도  
...       ...  
6529  제주특별자치도  
6705  제주특별자치도  
6759  제주특별자치도  
6760  제주특별자치도  
6797  제주특별자치도  

[4988 rows x 16 columns]

In [ ]:
# '지번주소' 열에서 '읍', '면', '동'이 들어가는 행 추출
df_6['읍면'] = df_6['지번주소'].str.split().str[2]
df_6

콘텐츠아이디  콘텐츠분류                         제목  \
0     CNTS_000000000021903     숙박                       해피펜션   
1     CONT_000000000501391    음식점                     탐라 뚝배기   
2     CONT_000000000501399    음식점                        토담골   
3     CNTS_000000000018301    음식점                       우유부단   
4     CNTS_000000000020277    관광지                        공천포   
...                    ...    ...                        ...   
6529  CNTS_200000000015288  축제/행사  JSW SPRING BREEZE CONCERT   
6705  CNTS_200000000015227    관광지                      한형수정원   
6759  CNTS_200000000015249    음식점                한형수정원 테쉬폰카페   
6760  CNTS_200000000014966     쇼핑                        유화당   
6797  CNTS_200000000015266    음식점                    미베이커리카페   

                              지번주소                           도로명주소         위도  \
0           제주특별자치도 제주시 외도2동 317-1              제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 서귀포시 표선면 표선리 879      제주특별자치도 서귀포시 표선면 표선백사로 127  33.326770   
2            제주특별자치도 제주시 아라이동 2425              제주특별자치도 제주시 간월동로 5  33.485904   
3        제주특별자치도 제주시 한림읍 금악리 142-2         제주특별자치도 제주시 한림읍 금악동길 38  33.347640   
4             제주특별자치도 서귀포시 남원읍 신례리            제주특별자치도 서귀포시 남원읍 신례리  33.267690   
...                            ...                             ...        ...   
6529  제주특별자치도 서귀포시 안덕면 서광리 산 35-16  제주특별자치도 서귀포시 안덕면 신화역사로304번길 98  33.294647   
6705      제주특별자치도 제주시 한림읍 협재리 1352         제주특별자치도 제주시 한림읍 옹포2길 44  33.397039   
6759      제주특별자치도 제주시 한림읍 협재리 1352         제주특별자치도 제주시 한림읍 옹포2길 44  33.397039   
6760        제주특별자치도 서귀포시 서귀동 481-1          제주특별자치도 서귀포시 태평로 409-2  33.245194   
6797          제주특별자치도 서귀포시 강정동 847           제주특별자치도 서귀포시 이어도로 780  33.236092   

              경도   평점    조회수  좋아요수  리뷰수  북마크수  소셜네트워크공유수     인기점수  사진공유수  \
0     126.433780  0.0    979     0    0     1          0   1958.0    0.0   
1     126.836690  0.0   3120     0    0    12          1   6240.0    0.0   
2     126.543496  0.0   1973     0    0     4          0   3946.0    0.0   
3     126.328260  5.0  21460    14   38   253          5  43152.0   51.0   
4     126.645996  4.0   4779     0    9    26          2   9603.0   18.0   
...          ...  ...    ...   ...  ...   ...        ...      ...    ...   
6529  126.314558  0.0    115     0    0     0          0    230.0    0.0   
6705  126.250255  0.0     92     0    0     0          0    184.0    0.0   
6759  126.250255  0.0     42     0    0     0          0     84.0    0.0   
6760  126.563893  0.0      0     0    0     0          0      0.0    0.0   
6797  126.498936  0.0     18     1    0     0          0     39.0    0.0   

            도    읍면  
0     제주특별자치도  외도2동  
1     제주특별자치도   표선면  
2     제주특별자치도  아라이동  
3     제주특별자치도   한림읍  
4     제주특별자치도   남원읍  
...       ...   ...  
6529  제주특별자치도   안덕면  
6705  제주특별자치도   한림읍  
6759  제주특별자치도   한림읍  
6760  제주특별자치도   서귀동  
6797  제주특별자치도   강정동  

[4988 rows x 17 columns]

In [ ]:
df_6['시'] = df['지번주소'].str.split().str[1]
df_6

콘텐츠아이디  콘텐츠분류                         제목  \
0     CNTS_000000000021903     숙박                       해피펜션   
1     CONT_000000000501391    음식점                     탐라 뚝배기   
2     CONT_000000000501399    음식점                        토담골   
3     CNTS_000000000018301    음식점                       우유부단   
4     CNTS_000000000020277    관광지                        공천포   
...                    ...    ...                        ...   
6529  CNTS_200000000015288  축제/행사  JSW SPRING BREEZE CONCERT   
6705  CNTS_200000000015227    관광지                      한형수정원   
6759  CNTS_200000000015249    음식점                한형수정원 테쉬폰카페   
6760  CNTS_200000000014966     쇼핑                        유화당   
6797  CNTS_200000000015266    음식점                    미베이커리카페   

                              지번주소                           도로명주소         위도  \
0           제주특별자치도 제주시 외도2동 317-1              제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 서귀포시 표선면 표선리 879      제주특별자치도 서귀포시 표선면 표선백사로 127  33.326770   
2            제주특별자치도 제주시 아라이동 2425              제주특별자치도 제주시 간월동로 5  33.485904   
3        제주특별자치도 제주시 한림읍 금악리 142-2         제주특별자치도 제주시 한림읍 금악동길 38  33.347640   
4             제주특별자치도 서귀포시 남원읍 신례리            제주특별자치도 서귀포시 남원읍 신례리  33.267690   
...                            ...                             ...        ...   
6529  제주특별자치도 서귀포시 안덕면 서광리 산 35-16  제주특별자치도 서귀포시 안덕면 신화역사로304번길 98  33.294647   
6705      제주특별자치도 제주시 한림읍 협재리 1352         제주특별자치도 제주시 한림읍 옹포2길 44  33.397039   
6759      제주특별자치도 제주시 한림읍 협재리 1352         제주특별자치도 제주시 한림읍 옹포2길 44  33.397039   
6760        제주특별자치도 서귀포시 서귀동 481-1          제주특별자치도 서귀포시 태평로 409-2  33.245194   
6797          제주특별자치도 서귀포시 강정동 847           제주특별자치도 서귀포시 이어도로 780  33.236092   

              경도   평점    조회수  좋아요수  리뷰수  북마크수  소셜네트워크공유수     인기점수  사진공유수  \
0     126.433780  0.0    979     0    0     1          0   1958.0    0.0   
1     126.836690  0.0   3120     0    0    12          1   6240.0    0.0   
2     126.543496  0.0   1973     0    0     4          0   3946.0    0.0   
3     126.328260  5.0  21460    14   38   253          5  43152.0   51.0   
4     126.645996  4.0   4779     0    9    26          2   9603.0   18.0   
...          ...  ...    ...   ...  ...   ...        ...      ...    ...   
6529  126.314558  0.0    115     0    0     0          0    230.0    0.0   
6705  126.250255  0.0     92     0    0     0          0    184.0    0.0   
6759  126.250255  0.0     42     0    0     0          0     84.0    0.0   
6760  126.563893  0.0      0     0    0     0          0      0.0    0.0   
6797  126.498936  0.0     18     1    0     0          0     39.0    0.0   

            도    읍면     시  
0     제주특별자치도  외도2동   제주시  
1     제주특별자치도   표선면  서귀포시  
2     제주특별자치도  아라이동   제주시  
3     제주특별자치도   한림읍   제주시  
4     제주특별자치도   남원읍  서귀포시  
...       ...   ...   ...  
6529  제주특별자치도   안덕면  서귀포시  
6705  제주특별자치도   한림읍   제주시  
6759  제주특별자치도   한림읍   제주시  
6760  제주특별자치도   서귀동  서귀포시  
6797  제주특별자치도   강정동  서귀포시  

[4988 rows x 18 columns]

In [ ]:
df_6['읍면'].nunique()

215

In [ ]:
df_6['읍면'].unique()

array(['외도2동', '표선면', '아라이동', '한림읍', '남원읍', '노형동', '애월읍', '안덕면', '조천읍',
       '건입동', '하효동', '연동', '성산읍', '오라2동', '구좌읍', '대포동', '상예동', '삼도2동',
       '이도이동', '신효동', '강정동', '서홍동', '서사로', '상효동', '오일장중길', '임항로', '삼도이동',
       '한경면', '추자면', '하원동', '이호2동', '도두이동', '오등동', '용담삼동', '이도2동', '도남동',
       '연삼로', '일도이동', '용담3동', '신서귀로', '중문동', '일도1동', '선덕로', '명동로', '색달동',
       '토평동', '화북1동', '도련일동', '중앙로1길', '법환동', '도두1동', '봉개동', '하예동', '대정읍',
       '도순동', '대포중앙로', '원노형로', '동홍동로', '중산간서로', '호근동', '해안동', '영실로',
       '일도2동', '도두일동', '우도면', '외도이동', '서귀동', '서광로2길', '도두2동', '칠십리로',
       '내도동', '삼양3동', '516로', '삼양일동', '보목동', '아라1동', '회수동', '아라일동',
       '용담2동', '이호일동', '도평동', '동홍동', '아라2동', '산록남로', '도련이동', '중정로62번길',
       '연북로', '삼도1동', '용강동', '월평동', '서호동', '삼양2동', '천제연로', '중정로', '원노형6길',
       '진군길', '용담일동', '이어도로', '이도1동', '일도일동', '번영로', '외도일동', '중문관광로',
       '삼양동', '이호이동', '삼양1동', '영평동', '수목원길', '이도일동', '천수로', '용담이동', '용문로',
       '동문동로', '화북일동', '일주서로43번길', '삼동길', '남성로', '오복', '용해로',

In [ ]:
#df.loc[df['특정열'] == '특정문자값', '특정열'] = '다른문자값'

df_6.loc[df_6['읍면'] == '외도2동', '읍면'] = '외도동'
df_6.loc[df_6['읍면'] == '아라이동', '읍면'] = '아라동'
df_6.loc[df_6['읍면'] == '오라2동', '읍면'] = '오라동'

df_6.loc[df_6['읍면'] == '서홍동', '읍면'] = '서흥동'
df_6.loc[df_6['읍면'] == '동홍동', '읍면'] = '동흥동'
df_6.loc[df_6['읍면'] == '삼도이동', '읍면'] = '삼도2동'
df_6.loc[df_6['읍면'] == '이호2동', '읍면'] = '이호동'
df_6.loc[df_6['읍면'] == '도두이동', '읍면'] = '도두동'
df_6.loc[df_6['읍면'] == '도두2동', '읍면'] = '도두동'
df_6.loc[df_6['읍면'] == '도두일동', '읍면'] = '도두동'
df_6.loc[df_6['읍면'] == '용담삼동', '읍면'] = '용담2동'
df_6.loc[df_6['읍면'] == '일도이동', '읍면'] = '일도2동'
df_6.loc[df_6['읍면'] == '도두1동', '읍면'] = '도두동'


df_6.loc[df_6['읍면'] == '화북1동', '읍면'] = '화북동'
df_6.loc[df_6['읍면'] == '화북2동', '읍면'] = '화북동'
df_6.loc[df_6['읍면'] == '화북일동', '읍면'] = '화북동'
df_6.loc[df_6['읍면'] == '화북이동', '읍면'] = '화북동'


df_6.loc[df_6['읍면'] == '삼양3동', '읍면'] = '삼양동'
df_6.loc[df_6['읍면'] == '삼양삼동', '읍면'] = '삼양동'
df_6.loc[df_6['읍면'] == '삼양2동', '읍면'] = '삼양동'
df_6.loc[df_6['읍면'] == '삼양이동', '읍면'] = '삼양동'
df_6.loc[df_6['읍면'] == '삼양1동', '읍면'] = '삼양동'
df_6.loc[df_6['읍면'] == '삼양일동', '읍면'] = '삼양동'


df_6.loc[df_6['읍면'] == '아라1동', '읍면'] = '아라동'
df_6.loc[df_6['읍면'] == '아라일동', '읍면'] = '아라동'
df_6.loc[df_6['읍면'] == '아라2동', '읍면'] = '아라동'
df_6.loc[df_6['읍면'] == '아라이동', '읍면'] = '아라동'
df_6.loc[df_6['읍면'] == '아라3동', '읍면'] = '아라동'
df_6.loc[df_6['읍면'] == '아라삼동', '읍면'] = '아라동'


df_6.loc[df_6['읍면'] == '용담일동', '읍면'] = '용담1동'
df_6.loc[df_6['읍면'] == '용담이동', '읍면'] = '용담2동'
df_6.loc[df_6['읍면'] == '용담삼동', '읍면'] = '용담2동'
df_6.loc[df_6['읍면'] == '용담3동', '읍면'] = '용담2동'


df_6.loc[df_6['읍면'] == '이호2동', '읍면'] = '이호동'
df_6.loc[df_6['읍면'] == '이호이동', '읍면'] = '이호동'
df_6.loc[df_6['읍면'] == '이호1동', '읍면'] = '이호동'
df_6.loc[df_6['읍면'] == '이호일동', '읍면'] = '이호동'


df_6.loc[df_6['읍면'] == '외도일동', '읍면'] = '외도동'
df_6.loc[df_6['읍면'] == '외도1동', '읍면'] = '외도동'
df_6.loc[df_6['읍면'] == '외도이동', '읍면'] = '외도동'
df_6.loc[df_6['읍면'] == '외도2동', '읍면'] = '외도동'


df_6.loc[df_6['읍면'] == '오라이동', '읍면'] = '오라동'
df_6.loc[df_6['읍면'] == '오라2동', '읍면'] = '오라동'
df_6.loc[df_6['읍면'] == '오라일동', '읍면'] = '오라동'
df_6.loc[df_6['읍면'] == '오라1동', '읍면'] = '오라동'
df_6.loc[df_6['읍면'] == '오라삼동', '읍면'] = '오라동'
df_6.loc[df_6['읍면'] == '오라3동', '읍면'] = '오라동'


df_6.loc[df_6['읍면'] == '삼도일동', '읍면'] = '삼도1동'
df_6.loc[df_6['읍면'] == '삼도이동', '읍면'] = '삼도2동'
df_6.loc[df_6['읍면'] == '삼도삼동', '읍면'] = '삼도2동'


df_6.loc[df_6['읍면'] == '추자동', '읍면'] = '추자면'
df_6.loc[df_6['읍면'] == '추자도', '읍면'] = '추자면'

df_6.loc[df_6['읍면'] == '이도일동', '읍면'] = '이도1동'
df_6.loc[df_6['읍면'] == '이도이동', '읍면'] = '이도2동'

df_6.loc[df_6['읍면'] == '성산면', '읍면'] = '성산읍'

# df_6.loc[df_6['읍면'] == '외도일동', '읍면'] = '외도동'




In [ ]:
df_6['읍면'].nunique()

176

In [ ]:
df_6['읍면'].unique()

array(['외도동', '표선면', '아라동', '한림읍', '남원읍', '노형동', '애월읍', '안덕면', '조천읍',
       '건입동', '하효동', '연동', '성산읍', '오라동', '구좌읍', '대포동', '상예동', '삼도2동',
       '이도2동', '신효동', '강정동', '서흥동', '서사로', '상효동', '오일장중길', '임항로', '한경면',
       '추자면', '하원동', '이호동', '도두동', '오등동', '용담2동', '도남동', '연삼로', '일도2동',
       '신서귀로', '중문동', '일도1동', '선덕로', '명동로', '색달동', '토평동', '화북동', '도련일동',
       '중앙로1길', '법환동', '봉개동', '하예동', '대정읍', '도순동', '대포중앙로', '원노형로',
       '동홍동로', '중산간서로', '호근동', '해안동', '영실로', '우도면', '서귀동', '서광로2길',
       '칠십리로', '내도동', '삼양동', '516로', '보목동', '회수동', '도평동', '동흥동', '산록남로',
       '도련이동', '중정로62번길', '연북로', '삼도1동', '용강동', '월평동', '서호동', '천제연로',
       '중정로', '원노형6길', '진군길', '용담1동', '이어도로', '이도1동', '일도일동', '번영로',
       '중문관광로', '영평동', '수목원길', '천수로', '용문로', '동문동로', '일주서로43번길', '삼동길',
       '남성로', '오복', '용해로', '관덕로', '선사로2길', '산지로', '1100로', '소보리당로',
       '서해안로', '남광북1길', '중앙로42번길', '진군3길', '색달중앙로121번길', '막숙포로', '동광로2길',
       '문송1길', '상예로', '한립읍', '내도7길', '동문로', '곽지1길', '은남4길', '삼성로', '탑동로',
      

In [ ]:
df_6['읍면'].shape

(4988,)

# 빌딩 비율 (읍면동 기준값) df_6 -> df_b 완


In [ ]:
df_6

In [ ]:
ratio = pd.read_csv('/content/B_ratio_data_v3_final.csv', encoding='cp949')
ratio

지역이름 지역이름.1  켭치는 영역 픽셀 수   모든 영역 픽셀 수   ratio
0     map-7gGg    구좌읍         76338      1356567   5.63
1       anduck    조천읍         84607      1078137   7.85
2          ara    한림읍         51344       662529   7.75
3      bonggae    한경면         43743       559125   7.82
4      cheonji    대정읍         45307       549426   8.25
5        chuja    안덕면         36615       741471   4.94
6     daecheon    남원읍         91750      1340817   6.84
7      daejung    표선면         53738       956304   5.62
8       daerun    성산읍         33638       788340   4.27
9         dodu    애월읍         84489      1453041   5.81
10    donghung    예래동         17131       276144   6.20
11        ewol    중문동         46074       446328  10.32
12        guja    대천동         37566       358797  10.47
13       gunip    영천동         27923       326979   8.54
14    hangyung    효돈동          5536        45567  12.15
15      hanlim    송산동          6242        31053  20.10
16     hwabuck    대륜동         31961       161139  19.83
17      hyodon    서흥동          5120        95919   5.34
18       ido11    천지동          1758        12996  13.53
19        ido2    정방동          2450         4017  60.99
20         iho    중앙동          2438         2505  97.33
21      ildo11    동흥동         22779       103473  22.01
22       ildo2    외도동         14537        63876  22.76
23     jocheon    이호동          7594        19515  38.91
24     jungang    도두동         10454        28332  36.90
25    jungbang   용담2동          5030        36783  13.67
26     jungmun    노형동         27511       328212   8.38
27      namwon     연동         28155        90825  31.00
28     nohyung    오라동         19469       205074   9.49
29         ora    아라동         35887       514215   6.98
30      pyosun    봉개동          8614       342663   2.51
31      samdo1    삼양동         20135        74463  27.04
32      samdo2    화북동         27208        61908  43.95
33     samyang   이도2동         22689        41112  55.19
34     seohung   용담1동          4003         4845  82.62
35     songsan   삼도1동          6073         6528  93.03
36     sungsan   삼도2동          4465         6474  68.97
37         udo   일도2동         14985        17343  86.40
38       weodo   일도1동          2244         2355  95.29
39    yeahyeah    건입동         11943        21633  55.21
40        yeon   이도1동          5580         5676  98.31
41    yongdam2    우도면          1143        50685   2.26
42  youngcheon    추자면          1434        65796   2.18

In [ ]:
ratio=ratio[['지역이름.1', 'ratio']]
ratio

지역이름.1  ratio
0     구좌읍   5.63
1     조천읍   7.85
2     한림읍   7.75
3     한경면   7.82
4     대정읍   8.25
5     안덕면   4.94
6     남원읍   6.84
7     표선면   5.62
8     성산읍   4.27
9     애월읍   5.81
10    예래동   6.20
11    중문동  10.32
12    대천동  10.47
13    영천동   8.54
14    효돈동  12.15
15    송산동  20.10
16    대륜동  19.83
17    서흥동   5.34
18    천지동  13.53
19    정방동  60.99
20    중앙동  97.33
21    동흥동  22.01
22    외도동  22.76
23    이호동  38.91
24    도두동  36.90
25   용담2동  13.67
26    노형동   8.38
27     연동  31.00
28    오라동   9.49
29    아라동   6.98
30    봉개동   2.51
31    삼양동  27.04
32    화북동  43.95
33   이도2동  55.19
34   용담1동  82.62
35   삼도1동  93.03
36   삼도2동  68.97
37   일도2동  86.40
38   일도1동  95.29
39    건입동  55.21
40   이도1동  98.31
41    우도면   2.26
42    추자면   2.18

In [ ]:
ratio.rename(columns={'지역이름.1': '읍면'}, inplace=True)

<ipython-input-185-ab0f893d3257>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ratio.rename(columns={'지역이름.1': '읍면'}, inplace=True)


In [ ]:
ratio

읍면  ratio
0    구좌읍   5.63
1    조천읍   7.85
2    한림읍   7.75
3    한경면   7.82
4    대정읍   8.25
5    안덕면   4.94
6    남원읍   6.84
7    표선면   5.62
8    성산읍   4.27
9    애월읍   5.81
10   예래동   6.20
11   중문동  10.32
12   대천동  10.47
13   영천동   8.54
14   효돈동  12.15
15   송산동  20.10
16   대륜동  19.83
17   서흥동   5.34
18   천지동  13.53
19   정방동  60.99
20   중앙동  97.33
21   동흥동  22.01
22   외도동  22.76
23   이호동  38.91
24   도두동  36.90
25  용담2동  13.67
26   노형동   8.38
27    연동  31.00
28   오라동   9.49
29   아라동   6.98
30   봉개동   2.51
31   삼양동  27.04
32   화북동  43.95
33  이도2동  55.19
34  용담1동  82.62
35  삼도1동  93.03
36  삼도2동  68.97
37  일도2동  86.40
38  일도1동  95.29
39   건입동  55.21
40  이도1동  98.31
41   우도면   2.26
42   추자면   2.18

43개의 읍면동

4988-> 4120


In [ ]:
df_b =pd.merge(df_6, ratio, how='inner', on='읍면')
df_b

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도   평점   조회수  좋아요수  리뷰수  북마크수  소셜네트워크공유수     인기점수  사진공유수  \
0     126.433780  0.0   979     0    0     1          0   1958.0    0.0   
1     126.424573  0.0  1043     0    0     1          0   2086.0    0.0   
2     126.429609  0.0  2118     0    0     1          0   4236.0    0.0   
3     126.436062  5.0  7851     0    1    14          8  15707.0    0.0   
4     126.436436  5.0  5153     0    3    47          0  10321.0    4.0   
...          ...  ...   ...   ...  ...   ...        ...      ...    ...   
4115  126.523442  0.0  2312     0    0     4          1   4624.0    0.0   
4116  126.529358  0.0   128     0    0     0          0    256.0    0.0   
4117  126.523939  0.0  1305     1    0     5          0   2613.0    0.0   
4118  126.527915  0.0  1153     0    0     1          2   2306.0    0.0   
4119  126.526077  0.0   194     0    0     5          0    388.0    0.0   

            도    읍면    시  ratio  
0     제주특별자치도   외도동  제주시  22.76  
1     제주특별자치도   외도동  제주시  22.76  
2     제주특별자치도   외도동  제주시  22.76  
3     제주특별자치도   외도동  제주시  22.76  
4     제주특별자치도   외도동  제주시  22.76  
...       ...   ...  ...    ...  
4115  제주특별자치도  이도1동  제주시  98.31  
4116  제주특별자치도  이도1동  제주시  98.31  
4117  제주특별자치도  이도1동  제주시  98.31  
4118  제주특별자치도  이도1동  제주시  98.31  
4119  제주특별자치도  이도1동  제주시  98.31  

[4120 rows x 19 columns]

In [ ]:
df_b['ratio'].nunique() # 사용되는 읍면 개수 34개

34

#관광객수 합치기 df_b -> df_7 완

In [ ]:
tourist = pd.read_csv("/content/제주도 관광객 수_2022년_취합.csv", encoding='cp949')
tour = tourist.iloc[:, 1:3]

In [ ]:
tour

행정동        관광객수
0    안덕면  11,308,881
1    성산읍   9,044,676
2    예래동   7,602,423
3    남원읍   4,769,707
4    대정읍   4,956,729
5    표선면   5,707,617
6    대륜동   4,363,931
7    중문동   4,572,786
8    중앙동   2,914,056
9    대천동   2,529,360
10   송산동   2,684,303
11   천지동   1,932,170
12   정방동   1,855,452
13   영천동   1,819,173
14   동흥동     899,817
15   효돈동   1,006,672
16   서흥동     460,614
17  용담2동  12,013,080
18   애월읍  13,178,838
19    연동  12,933,413
20   노형동   7,543,251
21   조천읍   9,102,603
22   구좌읍   8,197,921
23   한림읍   7,826,109
24   한경면   3,962,233
25  일도2동   3,188,022
26  이도2동   2,396,297
27   건입동   2,821,681
28   아라동   2,434,115
29  삼도2동   2,440,882
30   도두동   2,250,484
31   이호동   1,987,390
32   오라동   1,447,924
33   외도동   1,389,162
34   봉개동   1,800,716
35   우도면   1,710,982
36  이도1동     979,598
37  일도1동     907,667
38   삼양동     884,070
39   화북동     878,953
40  삼도1동     593,002
41   추자면     823,407
42  용담1동     476,474

In [ ]:
df_7 = pd.merge(df_b, tour, how='left', left_on="읍면", right_on="행정동")
df_7

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도   평점   조회수  좋아요수  ...  북마크수  소셜네트워크공유수     인기점수  사진공유수  \
0     126.433780  0.0   979     0  ...     1          0   1958.0    0.0   
1     126.424573  0.0  1043     0  ...     1          0   2086.0    0.0   
2     126.429609  0.0  2118     0  ...     1          0   4236.0    0.0   
3     126.436062  5.0  7851     0  ...    14          8  15707.0    0.0   
4     126.436436  5.0  5153     0  ...    47          0  10321.0    4.0   
...          ...  ...   ...   ...  ...   ...        ...      ...    ...   
4115  126.523442  0.0  2312     0  ...     4          1   4624.0    0.0   
4116  126.529358  0.0   128     0  ...     0          0    256.0    0.0   
4117  126.523939  0.0  1305     1  ...     5          0   2613.0    0.0   
4118  126.527915  0.0  1153     0  ...     1          2   2306.0    0.0   
4119  126.526077  0.0   194     0  ...     5          0    388.0    0.0   

            도    읍면    시  ratio   행정동       관광객수  
0     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
1     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
2     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
3     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
4     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
...       ...   ...  ...    ...   ...        ...  
4115  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  
4116  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  
4117  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  
4118  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  
4119  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  

[4120 rows x 21 columns]

In [ ]:
df_7['행정동'].isnull()

0       False
1       False
2       False
3       False
4       False
        ...  
4115    False
4116    False
4117    False
4118    False
4119    False
Name: 행정동, Length: 4120, dtype: bool

In [ ]:
df_7['관광객수'].isnull().sum() # 깔끔하게 완료

0

# 카드사용금액 합치기 df_7->df_10-> df_11 완

In [ ]:
df_10 = df_7.copy()
df_10

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도   평점   조회수  좋아요수  ...  북마크수  소셜네트워크공유수     인기점수  사진공유수  \
0     126.433780  0.0   979     0  ...     1          0   1958.0    0.0   
1     126.424573  0.0  1043     0  ...     1          0   2086.0    0.0   
2     126.429609  0.0  2118     0  ...     1          0   4236.0    0.0   
3     126.436062  5.0  7851     0  ...    14          8  15707.0    0.0   
4     126.436436  5.0  5153     0  ...    47          0  10321.0    4.0   
...          ...  ...   ...   ...  ...   ...        ...      ...    ...   
4115  126.523442  0.0  2312     0  ...     4          1   4624.0    0.0   
4116  126.529358  0.0   128     0  ...     0          0    256.0    0.0   
4117  126.523939  0.0  1305     1  ...     5          0   2613.0    0.0   
4118  126.527915  0.0  1153     0  ...     1          2   2306.0    0.0   
4119  126.526077  0.0   194     0  ...     5          0    388.0    0.0   

            도    읍면    시  ratio   행정동       관광객수  
0     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
1     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
2     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
3     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
4     제주특별자치도   외도동  제주시  22.76   외도동  1,389,162  
...       ...   ...  ...    ...   ...        ...  
4115  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  
4116  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  
4117  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  
4118  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  
4119  제주특별자치도  이도1동  제주시  98.31  이도1동    979,598  

[4120 rows x 21 columns]

In [ ]:
card = pd.read_csv('/content/카드사용금액(읍면동)_수정본.csv',encoding='utf-8')
card

읍면  카드사용금액
0    노형동   657.8
1     연동   347.2
2    예래동   214.1
3   용담2동   177.1
4    애월읍   173.6
5    안덕면   142.5
6    조천읍   134.3
7    성산읍   123.0
8    중문동   114.5
9    한림읍   109.7
10   표선면    99.4
11   구좌읍    87.7
12   대정읍    65.2
13   대륜동    60.8
14  이도2동    59.7
15   중앙동    54.3
16  삼도2동    52.1
17  일도1동    48.7
18   남원읍    45.4
19   도두동    44.6
20   대천동    40.3
21   한경면    35.7
22   정방동    34.2
23   우도면    31.7
24  이도1동    26.5
25   오라동    26.5
26   건입동    26.4
27   아라동    25.8
28   송산동    21.3
29   천지동    20.5
30   영천동    19.0
31   효돈동    19.0
32   서흥동    19.0
33   동흥동    19.0
34   외도동    19.0
35   이호동    19.0
36   봉개동    19.0
37   삼양동    19.0
38   화북동    19.0
39  용담1동    19.0
40  삼도1동    19.0
41  일도2동    19.0

In [ ]:
df_11 = pd.merge(df_10, card, how='left', on='읍면')
df_11

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도   평점   조회수  좋아요수  ...  소셜네트워크공유수     인기점수  사진공유수        도  \
0     126.433780  0.0   979     0  ...          0   1958.0    0.0  제주특별자치도   
1     126.424573  0.0  1043     0  ...          0   2086.0    0.0  제주특별자치도   
2     126.429609  0.0  2118     0  ...          0   4236.0    0.0  제주특별자치도   
3     126.436062  5.0  7851     0  ...          8  15707.0    0.0  제주특별자치도   
4     126.436436  5.0  5153     0  ...          0  10321.0    4.0  제주특별자치도   
...          ...  ...   ...   ...  ...        ...      ...    ...      ...   
4115  126.523442  0.0  2312     0  ...          1   4624.0    0.0  제주특별자치도   
4116  126.529358  0.0   128     0  ...          0    256.0    0.0  제주특별자치도   
4117  126.523939  0.0  1305     1  ...          0   2613.0    0.0  제주특별자치도   
4118  126.527915  0.0  1153     0  ...          2   2306.0    0.0  제주특별자치도   
4119  126.526077  0.0   194     0  ...          0    388.0    0.0  제주특별자치도   

        읍면    시  ratio   행정동       관광객수 카드사용금액  
0      외도동  제주시  22.76   외도동  1,389,162   19.0  
1      외도동  제주시  22.76   외도동  1,389,162   19.0  
2      외도동  제주시  22.76   외도동  1,389,162   19.0  
3      외도동  제주시  22.76   외도동  1,389,162   19.0  
4      외도동  제주시  22.76   외도동  1,389,162   19.0  
...    ...  ...    ...   ...        ...    ...  
4115  이도1동  제주시  98.31  이도1동    979,598   26.5  
4116  이도1동  제주시  98.31  이도1동    979,598   26.5  
4117  이도1동  제주시  98.31  이도1동    979,598   26.5  
4118  이도1동  제주시  98.31  이도1동    979,598   26.5  
4119  이도1동  제주시  98.31  이도1동    979,598   26.5  

[4120 rows x 22 columns]

In [ ]:
df_11['카드사용금액'].isnull().sum()

101

In [ ]:
df_11[df_11['카드사용금액'].isnull()]

콘텐츠아이디 콘텐츠분류         제목                        지번주소  \
3281  CNTS_000000000019091   관광지      묵리처녀당    제주특별자치도 제주시 추자면 묵리 229-1   
3282  CNTS_000000000019090   관광지     묵리해안쉼터    제주특별자치도 제주시 추자면 묵리 731-4   
3283  CNTS_000000000019095   관광지     참굴비조형물  제주특별자치도 제주시 추자면 묵리 산 131-1   
3284  CNTS_000000000019100   관광지      예초리포구   제주특별자치도 제주시 추자면 예초리 200-1   
3285  CNTS_000000000019104   관광지  추자도 숭어체험장  제주특별자치도 제주시 추자면 신양리 874-11   
...                    ...   ...        ...                         ...   
3377  CNTS_200000000009138    숙박       동일민박    제주특별자치도 제주시 추자면 대서리 4-14   
3378  CNTS_200000000009104    숙박    뉴아일랜드민박     제주특별자치도 제주시 추자면 신양리 417   
3379  CNTS_200000000009106   음식점   신등대수산회센타     제주특별자치도 제주시 추자면 대서리 5-7   
3380  CNTS_200000000009108    숙박    추자바다25시      제주특별자치도 제주시 추자면 묵리 280   
3381  CNTS_200000000009109   음식점       귀빈식당       제주특별자치도 제주시 추자면 대서리 6   

                           도로명주소         위도          경도   평점   조회수  좋아요수  ...  \
3281    제주특별자치도 제주시 추자면 묵리 229-1  33.943170  126.314260  0.0  2826     0  ...   
3282   제주특별자치도 제주시 추자면 추자로 376-6  33.944927  126.312162  0.0  2966     0  ...   
3283                         NaN  33.952576  126.304320  0.0  2545     1  ...   
3284     제주특별자치도 제주시 추자면 추자로 753  33.955759  126.332432  0.0  2771     1  ...   
3285      제주특별자치도 제주시 추자면 석두리길 4  33.939980  126.323540  5.0  4442     1  ...   
...                          ...        ...         ...  ...   ...   ...  ...   
3377  제주특별자치도 제주시 추자면 대서1길 9, 3층  33.964290  126.292986  0.0   411     0  ...   
3378   제주특별자치도 제주시 추자면 추자로 581-6  33.935434  126.322764  0.0   466     0  ...   
3379      제주특별자치도 제주시 추자면 추자로 12  33.963290  126.297689  0.0   470     0  ...   
3380     제주특별자치도 제주시 추자면 묵리2길 48  33.944852  126.315100  0.0  1015     0  ...   
3381    제주특별자치도 제주시 추자면 추자로 14-3  33.964290  126.292986  0.0   778     0  ...   

      소셜네트워크공유수    인기점수  사진공유수        도   읍면    시 ratio  행정동     관광객수 카드사용금액  
3281          0  5652.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
3282          0  5932.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
3283          1  5093.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
3284          0  5545.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
3285          1  8892.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
...         ...     ...    ...      ...  ...  ...   ...  ...      ...    ...  
3377          0   822.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
3378          0   932.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
3379          1   940.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
3380          1  2030.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  
3381          0  1556.0    0.0  제주특별자치도  추자면  제주시  2.18  추자면  823,407    NaN  

[101 rows x 22 columns]

추자면, 우도면 값도 19로 일괄지정

In [ ]:
df_11['카드사용금액'] = df_11['카드사용금액'].fillna(19)

In [ ]:
df_11['카드사용금액'].isna().sum()  #완

0

# 관광지 개수 df_11 -> df_12 완

In [ ]:
df_c = df_11.copy()
df_c

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도   평점   조회수  좋아요수  ...  소셜네트워크공유수     인기점수  사진공유수        도  \
0     126.433780  0.0   979     0  ...          0   1958.0    0.0  제주특별자치도   
1     126.424573  0.0  1043     0  ...          0   2086.0    0.0  제주특별자치도   
2     126.429609  0.0  2118     0  ...          0   4236.0    0.0  제주특별자치도   
3     126.436062  5.0  7851     0  ...          8  15707.0    0.0  제주특별자치도   
4     126.436436  5.0  5153     0  ...          0  10321.0    4.0  제주특별자치도   
...          ...  ...   ...   ...  ...        ...      ...    ...      ...   
4115  126.523442  0.0  2312     0  ...          1   4624.0    0.0  제주특별자치도   
4116  126.529358  0.0   128     0  ...          0    256.0    0.0  제주특별자치도   
4117  126.523939  0.0  1305     1  ...          0   2613.0    0.0  제주특별자치도   
4118  126.527915  0.0  1153     0  ...          2   2306.0    0.0  제주특별자치도   
4119  126.526077  0.0   194     0  ...          0    388.0    0.0  제주특별자치도   

        읍면    시  ratio   행정동       관광객수 카드사용금액  
0      외도동  제주시  22.76   외도동  1,389,162   19.0  
1      외도동  제주시  22.76   외도동  1,389,162   19.0  
2      외도동  제주시  22.76   외도동  1,389,162   19.0  
3      외도동  제주시  22.76   외도동  1,389,162   19.0  
4      외도동  제주시  22.76   외도동  1,389,162   19.0  
...    ...  ...    ...   ...        ...    ...  
4115  이도1동  제주시  98.31  이도1동    979,598   26.5  
4116  이도1동  제주시  98.31  이도1동    979,598   26.5  
4117  이도1동  제주시  98.31  이도1동    979,598   26.5  
4118  이도1동  제주시  98.31  이도1동    979,598   26.5  
4119  이도1동  제주시  98.31  이도1동    979,598   26.5  

[4120 rows x 22 columns]

In [ ]:
grouped_counts = df_c.groupby('읍면').size().reset_index(name='관광지개수')

In [ ]:
g = pd.DataFrame(grouped_counts)
g

읍면  관광지개수
0    건입동     57
1    구좌읍    439
2    남원읍    171
3    노형동    135
4    대정읍    171
5    도두동     46
6    동흥동     24
7    봉개동     28
8   삼도1동     15
9   삼도2동     63
10   삼양동     33
11   서흥동     50
12   성산읍    288
13   아라동     73
14   안덕면    335
15   애월읍    403
16    연동    187
17   오라동     54
18   외도동     25
19  용담1동     23
20  용담2동     77
21   우도면    134
22  이도1동     26
23  이도2동     92
24   이호동     33
25  일도1동     24
26  일도2동     29
27   조천읍    300
28   중문동     57
29   추자면    101
30   표선면    183
31   한경면    156
32   한림읍    270
33   화북동     18

In [ ]:
df_12= pd.merge(df_c,g , how='left', on='읍면')
df_12

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도   평점   조회수  좋아요수  ...     인기점수  사진공유수        도    읍면    시  \
0     126.433780  0.0   979     0  ...   1958.0    0.0  제주특별자치도   외도동  제주시   
1     126.424573  0.0  1043     0  ...   2086.0    0.0  제주특별자치도   외도동  제주시   
2     126.429609  0.0  2118     0  ...   4236.0    0.0  제주특별자치도   외도동  제주시   
3     126.436062  5.0  7851     0  ...  15707.0    0.0  제주특별자치도   외도동  제주시   
4     126.436436  5.0  5153     0  ...  10321.0    4.0  제주특별자치도   외도동  제주시   
...          ...  ...   ...   ...  ...      ...    ...      ...   ...  ...   
4115  126.523442  0.0  2312     0  ...   4624.0    0.0  제주특별자치도  이도1동  제주시   
4116  126.529358  0.0   128     0  ...    256.0    0.0  제주특별자치도  이도1동  제주시   
4117  126.523939  0.0  1305     1  ...   2613.0    0.0  제주특별자치도  이도1동  제주시   
4118  126.527915  0.0  1153     0  ...   2306.0    0.0  제주특별자치도  이도1동  제주시   
4119  126.526077  0.0   194     0  ...    388.0    0.0  제주특별자치도  이도1동  제주시   

      ratio   행정동       관광객수  카드사용금액 관광지개수  
0     22.76   외도동  1,389,162    19.0    25  
1     22.76   외도동  1,389,162    19.0    25  
2     22.76   외도동  1,389,162    19.0    25  
3     22.76   외도동  1,389,162    19.0    25  
4     22.76   외도동  1,389,162    19.0    25  
...     ...   ...        ...     ...   ...  
4115  98.31  이도1동    979,598    26.5    26  
4116  98.31  이도1동    979,598    26.5    26  
4117  98.31  이도1동    979,598    26.5    26  
4118  98.31  이도1동    979,598    26.5    26  
4119  98.31  이도1동    979,598    26.5    26  

[4120 rows x 23 columns]

# 유네스코 df_12 완

In [ ]:
df_12['유네스코유무']=0

In [ ]:
df_12

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도   평점   조회수  좋아요수  ...  사진공유수        도    읍면    시  ratio  \
0     126.433780  0.0   979     0  ...    0.0  제주특별자치도   외도동  제주시  22.76   
1     126.424573  0.0  1043     0  ...    0.0  제주특별자치도   외도동  제주시  22.76   
2     126.429609  0.0  2118     0  ...    0.0  제주특별자치도   외도동  제주시  22.76   
3     126.436062  5.0  7851     0  ...    0.0  제주특별자치도   외도동  제주시  22.76   
4     126.436436  5.0  5153     0  ...    4.0  제주특별자치도   외도동  제주시  22.76   
...          ...  ...   ...   ...  ...    ...      ...   ...  ...    ...   
4115  126.523442  0.0  2312     0  ...    0.0  제주특별자치도  이도1동  제주시  98.31   
4116  126.529358  0.0   128     0  ...    0.0  제주특별자치도  이도1동  제주시  98.31   
4117  126.523939  0.0  1305     1  ...    0.0  제주특별자치도  이도1동  제주시  98.31   
4118  126.527915  0.0  1153     0  ...    0.0  제주특별자치도  이도1동  제주시  98.31   
4119  126.526077  0.0   194     0  ...    0.0  제주특별자치도  이도1동  제주시  98.31   

       행정동       관광객수 카드사용금액  관광지개수 유네스코유무  
0      외도동  1,389,162   19.0     25      0  
1      외도동  1,389,162   19.0     25      0  
2      외도동  1,389,162   19.0     25      0  
3      외도동  1,389,162   19.0     25      0  
4      외도동  1,389,162   19.0     25      0  
...    ...        ...    ...    ...    ...  
4115  이도1동    979,598   26.5     26      0  
4116  이도1동    979,598   26.5     26      0  
4117  이도1동    979,598   26.5     26      0  
4118  이도1동    979,598   26.5     26      0  
4119  이도1동    979,598   26.5     26      0  

[4120 rows x 24 columns]

In [ ]:
# '제목' 열에서 'UNESCO'를 포함하는 행에 대해 '유네스코유무'를 1로 설정
df_12.loc[df_12['제목'].str.contains('UNESCO', case=False, na=False), '유네스코유무'] = 1
df_12

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도   평점   조회수  좋아요수  ...  사진공유수        도    읍면    시  ratio  \
0     126.433780  0.0   979     0  ...    0.0  제주특별자치도   외도동  제주시  22.76   
1     126.424573  0.0  1043     0  ...    0.0  제주특별자치도   외도동  제주시  22.76   
2     126.429609  0.0  2118     0  ...    0.0  제주특별자치도   외도동  제주시  22.76   
3     126.436062  5.0  7851     0  ...    0.0  제주특별자치도   외도동  제주시  22.76   
4     126.436436  5.0  5153     0  ...    4.0  제주특별자치도   외도동  제주시  22.76   
...          ...  ...   ...   ...  ...    ...      ...   ...  ...    ...   
4115  126.523442  0.0  2312     0  ...    0.0  제주특별자치도  이도1동  제주시  98.31   
4116  126.529358  0.0   128     0  ...    0.0  제주특별자치도  이도1동  제주시  98.31   
4117  126.523939  0.0  1305     1  ...    0.0  제주특별자치도  이도1동  제주시  98.31   
4118  126.527915  0.0  1153     0  ...    0.0  제주특별자치도  이도1동  제주시  98.31   
4119  126.526077  0.0   194     0  ...    0.0  제주특별자치도  이도1동  제주시  98.31   

       행정동       관광객수 카드사용금액  관광지개수 유네스코유무  
0      외도동  1,389,162   19.0     25      0  
1      외도동  1,389,162   19.0     25      0  
2      외도동  1,389,162   19.0     25      0  
3      외도동  1,389,162   19.0     25      0  
4      외도동  1,389,162   19.0     25      0  
...    ...        ...    ...    ...    ...  
4115  이도1동    979,598   26.5     26      0  
4116  이도1동    979,598   26.5     26      0  
4117  이도1동    979,598   26.5     26      0  
4118  이도1동    979,598   26.5     26      0  
4119  이도1동    979,598   26.5     26      0  

[4120 rows x 24 columns]

In [ ]:
df_12[df_12['제목'].str.contains('UNESCO', case=False, na=False)] #완

콘텐츠아이디 콘텐츠분류                    제목  \
1696  CONT_000000000500034   관광지   거문오름(UNESCO 세계자연유산)   
2306  CONT_000000000500349   관광지  성산일출봉(UNESCO 세계자연유산)   

                         지번주소                        도로명주소         위도  \
1696  제주특별자치도 제주시 조천읍 선흘리 478   제주특별자치도 제주시 조천읍 선교로 569-36  33.453073   
2306   제주특별자치도 서귀포시 성산읍 성산리 1  제주특별자치도 서귀포시 성산읍 일출로 284-12  33.462147   

              경도   평점     조회수  좋아요수  ...  사진공유수        도   읍면     시  ratio  \
1696  126.723508  4.0  204857    33  ...   61.0  제주특별자치도  조천읍   제주시   7.85   
2306  126.936424  5.0  611219   328  ...  978.0  제주특별자치도  성산읍  서귀포시   4.27   

      행정동       관광객수 카드사용금액  관광지개수 유네스코유무  
1696  조천읍  9,102,603  134.3    300      1  
2306  성산읍  9,044,676  123.0    288      1  

[2 rows x 24 columns]

In [ ]:
# '제목' 열에서 '한라산'를 포함하는 행을 선택
df_12[df_12['제목'].str.contains('한라산', case=False, na=False)]

콘텐츠아이디  콘텐츠분류               제목  \
200   CNTS_200000000014107    음식점  한라산아래첫마을 제주민속촌점   
242   CNTS_000000000022416    음식점        한라산빵 1950   
496   CNTS_200000000008050    관광지          한라산소주공장   
563   CNTS_000000000021906  축제/행사       한라산청정고사리축제   
710   CNTS_200000000011513  축제/행사  한라산 아래 놈삐꽃 필 무렵   
1504  CNTS_200000000007658    관광지      한라산 아래 첫 마을   
2890  CNTS_200000000012675    음식점            카페한라산   

                             지번주소                          도로명주소         위도  \
200    제주특별자치도 서귀포시 표선면 표선리 40-77  제주특별자치도 서귀포시 표선면 민속해안로 631-46  33.324332   
242       제주특별자치도 제주시 아라1동 2408-1             제주특별자치도 제주시 구산로 28  33.482927   
496       제주특별자치도 제주시 한림읍 옹포리 396        제주특별자치도 제주시 한림읍 한림로 555  33.407365   
563   제주특별자치도 서귀포시 남원읍 한남리 산 76-7       제주특별자치도 서귀포시 남원읍 서성로 788  33.330063   
710     제주특별자치도 서귀포시 남원읍 하례리 1809                            NaN  33.316389   
1504   제주특별자치도 서귀포시 안덕면 광평리 195-2        제주특별자치도 서귀포시 안덕면 광평로 41  33.329591   
2890   제주특별자치도 제주시 구좌읍 하도리 3200-9        제주특별자치도 제주시 구좌읍 면수1길 48  33.524677   

              경도   평점    조회수  좋아요수  ...     인기점수  사진공유수        도   읍면     시  \
200   126.843180  0.0    220     0  ...    440.0    0.0  제주특별자치도  표선면  서귀포시   
242   126.538268  5.0   3278     2  ...   6567.0    0.0  제주특별자치도  아라동   제주시   
496   126.256334  0.0  14863     1  ...  29729.0    0.0  제주특별자치도  한림읍   제주시   
563   126.677410  0.0   8343     1  ...  16689.0    0.0  제주특별자치도  남원읍  서귀포시   
710   126.597400  0.0    675     0  ...   1350.0    0.0  제주특별자치도  남원읍  서귀포시   
1504  126.381331  0.0   4363     1  ...   8729.0    0.0  제주특별자치도  안덕면  서귀포시   
2890  126.863073  0.0    402     1  ...    807.0    0.0  제주특별자치도  구좌읍   제주시   

     ratio  행정동        관광객수  카드사용금액 관광지개수  
200   5.62  표선면   5,707,617    99.4   183  
242   6.98  아라동   2,434,115    25.8    73  
496   7.75  한림읍   7,826,109   109.7   270  
563   6.84  남원읍   4,769,707    45.4   171  
710   6.84  남원읍   4,769,707    45.4   171  
1504  4.94  안덕면  11,308,881   142.5   335  
2890  5.63  구좌읍   8,197,921    87.7   439  

[7 rows x 23 columns]

한라산은 넣을 값이 없어서 생략

# df_12수정

In [ ]:
import pandas as pd


# CSV 파일 읽어오기
df = pd.read_csv('/content/df_12_수정.csv', encoding='cp949')
df

콘텐츠아이디  콘텐츠분류                      제목  \
0     CNTS_000000000021903     숙박                    해피펜션   
1     CNTS_000000000019096     숙박                  하늘타리펜션   
2     CONT_000000000501263    음식점                    사방팔방   
3     CONT_000000000500492    관광지                     월대천   
4     CONT_000000000501212    음식점                 돈사촌 외도점   
...                    ...    ...                     ...   
4115  CNTS_200000000008021     쇼핑                에브리바디빈티지   
4116  CNTS_200000000009374  축제/행사  삼성혈 문화의 거리 공연 모흥골 그디글락   
4117  CNTS_200000000012471    관광지                    고요산책   
4118  CNTS_200000000009859    음식점                    감초식당   
4119  CNTS_200000000012612    음식점                  야시장맛통령   

                           지번주소                   도로명주소         위도  \
0        제주특별자치도 제주시 외도2동 317-1      제주특별자치도 제주시 월대1길 4  33.494110   
1         제주특별자치도 제주시 외도이동 1991    제주특별자치도 제주시 연대마을길 59  33.493854   
2        제주특별자치도 제주시 외도일동 494-5   제주특별자치도 제주시 우정로10길 12  33.488888   
3        제주특별자치도 제주시 외도2동 232-1      제주특별자치도 제주시 월대6길 5  33.493509   
4        제주특별자치도 제주시 외도일동 125-8     제주특별자치도 제주시 우평로 283  33.483951   
...                         ...                     ...        ...   
4115    제주특별자치도 제주시 이도1동 1493-1    제주특별자치도 제주시 관덕로8길 31  33.510502   
4116  제주특별자치도 제주시 이도일동 1313 삼성혈  제주특별자치도 제주시 삼성로 22 삼성혈  33.504703   
4117    제주특별자치도 제주시 이도일동 1387-3    제주특별자치도 제주시 중앙로12길 5  33.511806   
4118    제주특별자치도 제주시 이도일동 1289-5    제주특별자치도 제주시 동광로1길 32  33.503294   
4119      제주특별자치도 제주시 이도일동 1434   제주특별자치도 제주시 관덕로14길 16  33.511714   

              경도  평점   조회수  좋아요수  ...   인기점수  사진공유수        도    읍면    시  \
0     126.433780   0   979     0  ...   1958      0  제주특별자치도   외도동  제주시   
1     126.424573   0  1043     0  ...   2086      0  제주특별자치도   외도동  제주시   
2     126.429609   0  2118     0  ...   4236      0  제주특별자치도   외도동  제주시   
3     126.436062   5  7851     0  ...  15707      0  제주특별자치도   외도동  제주시   
4     126.436436   5  5153     0  ...  10321      4  제주특별자치도   외도동  제주시   
...          ...  ..   ...   ...  ...    ...    ...      ...   ...  ...   
4115  126.523442   0  2312     0  ...   4624      0  제주특별자치도  이도1동  제주시   
4116  126.529358   0   128     0  ...    256      0  제주특별자치도  이도1동  제주시   
4117  126.523939   0  1305     1  ...   2613      0  제주특별자치도  이도1동  제주시   
4118  126.527915   0  1153     0  ...   2306      0  제주특별자치도  이도1동  제주시   
4119  126.526077   0   194     0  ...    388      0  제주특별자치도  이도1동  제주시   

      ratio       관광객수 카드사용금액  관광지개수 유네스코유무  
0     22.76  1,389,162   19.0     25      0  
1     22.76  1,389,162   19.0     25      0  
2     22.76  1,389,162   19.0     25      0  
3     22.76  1,389,162   19.0     25      0  
4     22.76  1,389,162   19.0     25      0  
...     ...        ...    ...    ...    ...  
4115  98.31    979,598   26.5     26      0  
4116  98.31    979,598   26.5     26      0  
4117  98.31    979,598   26.5     26      0  
4118  98.31    979,598   26.5     26      0  
4119  98.31    979,598   26.5     26      0  

[4120 rows x 23 columns]

In [ ]:
area = pd.read_csv('관광지면적.csv', encoding='cp949')
area

콘텐츠아이디                 제목                       지번주소  \
0     CNTS_000000000021903               해피펜션     제주특별자치도 제주시 외도2동 317-1   
1     CONT_000000000501391             탐라 뚝배기   제주특별자치도 서귀포시 표선면 표선리 879   
2     CONT_000000000501399                토담골      제주특별자치도 제주시 아라이동 2425   
3     CNTS_000000000018301               우유부단  제주특별자치도 제주시 한림읍 금악리 142-2   
4     CNTS_000000000020277                공천포       제주특별자치도 서귀포시 남원읍 신례리   
...                    ...                ...                        ...   
6794  CNTS_200000000015151        제주패들보드 서핑클럽                        NaN   
6795  CNTS_200000000015157          카카오패밀리 세화                        NaN   
6796  CNTS_200000000015164          강정 어촌체험마을                        NaN   
6797  CNTS_200000000015266            미베이커리카페       제주특별자치도 서귀포시 강정동 847   
6798  CNTS_200000000015306  제주마을산책 <서귀포 로컬여행>                        NaN   

          관광지면적  
0        2546.0  
1         876.0  
2        3230.0  
3        3708.0  
4     3252595.0  
...         ...  
6794        NaN  
6795        NaN  
6796        NaN  
6797        NaN  
6798        NaN  

[6799 rows x 4 columns]

In [ ]:
area['관광지면적'].isna().sum()

3219

In [ ]:
df_INNER_JOIN = pd.merge(dataframe_A, dataframe_B, left_on='콘텐츠아이디', right_on='콘텐츠아이디', how='inner')

In [ ]:
df_com = pd.merge(df, area, left_on='콘텐츠아이디', right_on='콘텐츠아이디', how='inner')

In [ ]:
df_com.isnull().sum()

콘텐츠아이디          0
콘텐츠분류           0
제목_x            0
지번주소_x          0
도로명주소          51
위도              0
경도              0
평점              0
조회수             0
좋아요수            0
리뷰수             0
북마크수            0
소셜네트워크공유수       0
인기점수            0
사진공유수           0
도               0
읍면              0
시               0
ratio           0
관광객수            0
카드사용금액          0
관광지개수           0
유네스코유무          0
제목_y            0
지번주소_y          0
관광지면적        1636
dtype: int64

In [ ]:
df_com['관광지면적'].dropna(inplace=True)

In [ ]:
df_com.isnull().sum()

콘텐츠아이디          0
콘텐츠분류           0
제목_x            0
지번주소_x          0
도로명주소          51
위도              0
경도              0
평점              0
조회수             0
좋아요수            0
리뷰수             0
북마크수            0
소셜네트워크공유수       0
인기점수            0
사진공유수           0
도               0
읍면              0
시               0
ratio           0
관광객수            0
카드사용금액          0
관광지개수           0
유네스코유무          0
제목_y            0
지번주소_y          0
관광지면적        1636
dtype: int64

In [ ]:
df_com = df_com[df_com['관광지면적'].isnull()==False]

In [ ]:
df_com.isnull().sum()

콘텐츠아이디        0
콘텐츠분류         0
제목_x          0
지번주소_x        0
도로명주소        33
위도            0
경도            0
평점            0
조회수           0
좋아요수          0
리뷰수           0
북마크수          0
소셜네트워크공유수     0
인기점수          0
사진공유수         0
도             0
읍면            0
시             0
ratio         0
관광객수          0
카드사용금액        0
관광지개수         0
유네스코유무        0
제목_y          0
지번주소_y        0
관광지면적         0
dtype: int64

In [ ]:
df_com.drop('도로명주소', axis=1, inplace=True)


In [ ]:
df_com.isnull().sum()

콘텐츠아이디       0
콘텐츠분류        0
제목_x         0
지번주소_x       0
위도           0
경도           0
평점           0
조회수          0
좋아요수         0
리뷰수          0
북마크수         0
소셜네트워크공유수    0
인기점수         0
사진공유수        0
도            0
읍면           0
시            0
ratio        0
관광객수         0
카드사용금액       0
관광지개수        0
유네스코유무       0
제목_y         0
지번주소_y       0
관광지면적        0
dtype: int64

In [ ]:
df_com.to_csv('jeju_data.csv', encoding='cp949', index=False)

데이터 순서바꾸기

In [ ]:
df_com

콘텐츠아이디  콘텐츠분류                      제목_x  \
0     CNTS_000000000021903     숙박                      해피펜션   
1     CNTS_000000000019096     숙박                    하늘타리펜션   
2     CONT_000000000501263    음식점                      사방팔방   
3     CONT_000000000500492    관광지                       월대천   
4     CONT_000000000501212    음식점                   돈사촌 외도점   
...                    ...    ...                       ...   
4109  CONT_000000000501232    음식점                 만세국수 삼성혈점   
4110  CONT_000000000500459    관광지                       오현단   
4111  CNTS_000000000021007     숙박                 비지터게스트하우스   
4113  CNTS_200000000007258  축제/행사   한 여름밤에 즐기는 박물관 미술관 프로그램   
4115  CNTS_200000000008021     쇼핑                  에브리바디빈티지   

                        지번주소_x         위도          경도  평점   조회수  좋아요수  리뷰수  \
0       제주특별자치도 제주시 외도2동 317-1  33.494110  126.433780   0   979     0    0   
1        제주특별자치도 제주시 외도이동 1991  33.493854  126.424573   0  1043     0    0   
2       제주특별자치도 제주시 외도일동 494-5  33.488888  126.429609   0  2118     0    0   
3       제주특별자치도 제주시 외도2동 232-1  33.493509  126.436062   5  7851     0    1   
4       제주특별자치도 제주시 외도일동 125-8  33.483951  126.436436   5  5153     0    3   
...                        ...        ...         ...  ..   ...   ...  ...   
4109   제주특별자치도 제주시 이도일동 1272-4  33.506063  126.529090   5  3640     1    1   
4110   제주특별자치도 제주시 이도1동 1421-3  33.510136  126.525620   0  8418     2    0   
4111  제주특별자치도 제주시 이도1동 1349-24  33.512360  126.525215   0  1998     0    0   
4113   제주특별자치도 제주시 이도1동 1244-4  33.503182  126.534082   0  2014     1    0   
4115   제주특별자치도 제주시 이도1동 1493-1  33.510502  126.523442   0  2312     0    0   

      ...    읍면    시  ratio       관광객수 카드사용금액 관광지개수 유네스코유무  \
0     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
1     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
2     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
3     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
4     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
...   ...   ...  ...    ...        ...    ...   ...    ...   
4109  ...  이도1동  제주시  98.31    979,598   26.5    26      0   
4110  ...  이도1동  제주시  98.31    979,598   26.5    26      0   
4111  ...  이도1동  제주시  98.31    979,598   26.5    26      0   
4113  ...  이도1동  제주시  98.31    979,598   26.5    26      0   
4115  ...  이도1동  제주시  98.31    979,598   26.5    26      0   

                          제목_y                    지번주소_y     관광지면적  
0                         해피펜션    제주특별자치도 제주시 외도2동 317-1    2546.0  
1                       하늘타리펜션     제주특별자치도 제주시 외도이동 1991   21305.0  
2                         사방팔방    제주특별자치도 제주시 외도일동 494-5    1064.3  
3                          월대천    제주특별자치도 제주시 외도2동 232-1  123440.0  
4                      돈사촌 외도점    제주특별자치도 제주시 외도일동 125-8    6452.0  
...                        ...                       ...       ...  
4109                 만세국수 삼성혈점   제주특별자치도 제주시 이도일동 1272-4    1064.3  
4110                       오현단   제주특별자치도 제주시 이도1동 1421-3    6238.0  
4111                 비지터게스트하우스  제주특별자치도 제주시 이도1동 1349-24  262215.0  
4113   한 여름밤에 즐기는 박물관 미술관 프로그램   제주특별자치도 제주시 이도1동 1244-4     357.0  
4115                  에브리바디빈티지   제주특별자치도 제주시 이도1동 1493-1     749.0  

[2484 rows x 25 columns]

In [ ]:
df_com2 = df_com[['콘텐츠아이디','관광객수','관광지면적','관광지개수','ratio','카드사용금액','조회수','평점','좋아요수','리뷰수','북마크수','소셜네트워크공유수','인기점수','사진공유수','콘텐츠분류','유네스코유무' ]]

In [ ]:
df_com2

콘텐츠아이디       관광객수     관광지면적  관광지개수  ratio  카드사용금액   조회수  \
0     CNTS_000000000021903  1,389,162    2546.0     25  22.76    19.0   979   
1     CNTS_000000000019096  1,389,162   21305.0     25  22.76    19.0  1043   
2     CONT_000000000501263  1,389,162    1064.3     25  22.76    19.0  2118   
3     CONT_000000000500492  1,389,162  123440.0     25  22.76    19.0  7851   
4     CONT_000000000501212  1,389,162    6452.0     25  22.76    19.0  5153   
...                    ...        ...       ...    ...    ...     ...   ...   
4109  CONT_000000000501232    979,598    1064.3     26  98.31    26.5  3640   
4110  CONT_000000000500459    979,598    6238.0     26  98.31    26.5  8418   
4111  CNTS_000000000021007    979,598  262215.0     26  98.31    26.5  1998   
4113  CNTS_200000000007258    979,598     357.0     26  98.31    26.5  2014   
4115  CNTS_200000000008021    979,598     749.0     26  98.31    26.5  2312   

      평점  좋아요수  리뷰수  북마크수  소셜네트워크공유수   인기점수  사진공유수  콘텐츠분류  유네스코유무  
0      0     0    0     1          0   1958      0     숙박       0  
1      0     0    0     1          0   2086      0     숙박       0  
2      0     0    0     1          0   4236      0    음식점       0  
3      5     0    1    14          8  15707      0    관광지       0  
4      5     0    3    47          0  10321      4    음식점       0  
...   ..   ...  ...   ...        ...    ...    ...    ...     ...  
4109   5     1    1    17          1   7288      5    음식점       0  
4110   0     2    0     4          2  16842      0    관광지       0  
4111   0     0    0     4          0   3996      0     숙박       0  
4113   0     1    0     2          3   4031      0  축제/행사       0  
4115   0     0    0     4          1   4624      0     쇼핑       0  

[2484 rows x 16 columns]

In [ ]:
df_com2.to_csv('jeju_data2.csv', encoding='cp949', index=False)

In [ ]:
import pandas as pd
from sklearn.preprocessing import OneHotEncoder

# OneHotEncoder 객체 생성
encoder = OneHotEncoder()

# 'Color' 열에 대해 원핫인코딩 적용
encoded_data = encoder.fit_transform(df_com2[['콘텐츠분류']]).toarray()

# 새로운 데이터프레임 생성
encoded_df = pd.DataFrame(encoded_data, columns=encoder.get_feature_names_out(['콘텐츠분류']))

# 기존 데이터프레임과 합치기
df_com3 = pd.concat([df_com2, encoded_df], axis=1)


In [ ]:
df_com3

콘텐츠아이디       관광객수     관광지면적  관광지개수  ratio  카드사용금액     조회수  \
0     CNTS_000000000021903  1,389,162    2546.0   25.0  22.76    19.0   979.0   
1     CNTS_000000000019096  1,389,162   21305.0   25.0  22.76    19.0  1043.0   
2     CONT_000000000501263  1,389,162    1064.3   25.0  22.76    19.0  2118.0   
3     CONT_000000000500492  1,389,162  123440.0   25.0  22.76    19.0  7851.0   
4     CONT_000000000501212  1,389,162    6452.0   25.0  22.76    19.0  5153.0   
...                    ...        ...       ...    ...    ...     ...     ...   
2477                   NaN        NaN       NaN    NaN    NaN     NaN     NaN   
2478                   NaN        NaN       NaN    NaN    NaN     NaN     NaN   
2479                   NaN        NaN       NaN    NaN    NaN     NaN     NaN   
2480                   NaN        NaN       NaN    NaN    NaN     NaN     NaN   
2483                   NaN        NaN       NaN    NaN    NaN     NaN     NaN   

       평점  좋아요수  리뷰수  ...     인기점수  사진공유수  콘텐츠분류  유네스코유무 콘텐츠분류_관광지  콘텐츠분류_쇼핑  \
0     0.0   0.0  0.0  ...   1958.0    0.0     숙박     0.0       0.0       0.0   
1     0.0   0.0  0.0  ...   2086.0    0.0     숙박     0.0       0.0       0.0   
2     0.0   0.0  0.0  ...   4236.0    0.0    음식점     0.0       0.0       0.0   
3     5.0   0.0  1.0  ...  15707.0    0.0    관광지     0.0       1.0       0.0   
4     5.0   0.0  3.0  ...  10321.0    4.0    음식점     0.0       0.0       0.0   
...   ...   ...  ...  ...      ...    ...    ...     ...       ...       ...   
2477  NaN   NaN  NaN  ...      NaN    NaN    NaN     NaN       0.0       1.0   
2478  NaN   NaN  NaN  ...      NaN    NaN    NaN     NaN       0.0       0.0   
2479  NaN   NaN  NaN  ...      NaN    NaN    NaN     NaN       0.0       0.0   
2480  NaN   NaN  NaN  ...      NaN    NaN    NaN     NaN       1.0       0.0   
2483  NaN   NaN  NaN  ...      NaN    NaN    NaN     NaN       0.0       1.0   

      콘텐츠분류_숙박  콘텐츠분류_음식점  콘텐츠분류_축제/행사  콘텐츠분류_테마여행  
0          1.0        0.0          0.0         0.0  
1          1.0        0.0          0.0         0.0  
2          0.0        1.0          0.0         0.0  
3          0.0        0.0          0.0         0.0  
4          0.0        1.0          0.0         0.0  
...        ...        ...          ...         ...  
2477       0.0        0.0          0.0         0.0  
2478       0.0        1.0          0.0         0.0  
2479       0.0        1.0          0.0         0.0  
2480       0.0        0.0          0.0         0.0  
2483       0.0        0.0          0.0         0.0  

[3510 rows x 22 columns]

# 수정된 ratio 합치기

In [ ]:
import pandas as pd

# CSV 파일 경로 설정 (파일명은 실제 파일이 위치한 경로에 맞게 변경해주세요)
csv_file_path = 'jeju_data.csv'

# CSV 파일 읽어오기
df = pd.read_csv(csv_file_path,encoding='cp949')

# DataFrame 출력
df

콘텐츠아이디  콘텐츠분류                      제목_x  \
0     CNTS_000000000021903     숙박                      해피펜션   
1     CNTS_000000000019096     숙박                    하늘타리펜션   
2     CONT_000000000501263    음식점                      사방팔방   
3     CONT_000000000500492    관광지                       월대천   
4     CONT_000000000501212    음식점                   돈사촌 외도점   
...                    ...    ...                       ...   
2479  CONT_000000000501232    음식점                 만세국수 삼성혈점   
2480  CONT_000000000500459    관광지                       오현단   
2481  CNTS_000000000021007     숙박                 비지터게스트하우스   
2482  CNTS_200000000007258  축제/행사   한 여름밤에 즐기는 박물관 미술관 프로그램   
2483  CNTS_200000000008021     쇼핑                  에브리바디빈티지   

                        지번주소_x         위도          경도  평점   조회수  좋아요수  리뷰수  \
0       제주특별자치도 제주시 외도2동 317-1  33.494110  126.433780   0   979     0    0   
1        제주특별자치도 제주시 외도이동 1991  33.493854  126.424573   0  1043     0    0   
2       제주특별자치도 제주시 외도일동 494-5  33.488888  126.429609   0  2118     0    0   
3       제주특별자치도 제주시 외도2동 232-1  33.493509  126.436062   5  7851     0    1   
4       제주특별자치도 제주시 외도일동 125-8  33.483951  126.436436   5  5153     0    3   
...                        ...        ...         ...  ..   ...   ...  ...   
2479   제주특별자치도 제주시 이도일동 1272-4  33.506063  126.529090   5  3640     1    1   
2480   제주특별자치도 제주시 이도1동 1421-3  33.510136  126.525620   0  8418     2    0   
2481  제주특별자치도 제주시 이도1동 1349-24  33.512360  126.525215   0  1998     0    0   
2482   제주특별자치도 제주시 이도1동 1244-4  33.503182  126.534082   0  2014     1    0   
2483   제주특별자치도 제주시 이도1동 1493-1  33.510502  126.523442   0  2312     0    0   

      ...    읍면    시  ratio       관광객수 카드사용금액 관광지개수 유네스코유무  \
0     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
1     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
2     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
3     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
4     ...   외도동  제주시  22.76  1,389,162   19.0    25      0   
...   ...   ...  ...    ...        ...    ...   ...    ...   
2479  ...  이도1동  제주시  98.31    979,598   26.5    26      0   
2480  ...  이도1동  제주시  98.31    979,598   26.5    26      0   
2481  ...  이도1동  제주시  98.31    979,598   26.5    26      0   
2482  ...  이도1동  제주시  98.31    979,598   26.5    26      0   
2483  ...  이도1동  제주시  98.31    979,598   26.5    26      0   

                          제목_y                    지번주소_y     관광지면적  
0                         해피펜션    제주특별자치도 제주시 외도2동 317-1    2546.0  
1                       하늘타리펜션     제주특별자치도 제주시 외도이동 1991   21305.0  
2                         사방팔방    제주특별자치도 제주시 외도일동 494-5    1064.3  
3                          월대천    제주특별자치도 제주시 외도2동 232-1  123440.0  
4                      돈사촌 외도점    제주특별자치도 제주시 외도일동 125-8    6452.0  
...                        ...                       ...       ...  
2479                 만세국수 삼성혈점   제주특별자치도 제주시 이도일동 1272-4    1064.3  
2480                       오현단   제주특별자치도 제주시 이도1동 1421-3    6238.0  
2481                 비지터게스트하우스  제주특별자치도 제주시 이도1동 1349-24  262215.0  
2482   한 여름밤에 즐기는 박물관 미술관 프로그램   제주특별자치도 제주시 이도1동 1244-4     357.0  
2483                  에브리바디빈티지   제주특별자치도 제주시 이도1동 1493-1     749.0  

[2484 rows x 25 columns]

In [ ]:
del df['ratio']

In [ ]:
df

콘텐츠아이디  콘텐츠분류                      제목_x  \
0     CNTS_000000000021903     숙박                      해피펜션   
1     CNTS_000000000019096     숙박                    하늘타리펜션   
2     CONT_000000000501263    음식점                      사방팔방   
3     CONT_000000000500492    관광지                       월대천   
4     CONT_000000000501212    음식점                   돈사촌 외도점   
...                    ...    ...                       ...   
2479  CONT_000000000501232    음식점                 만세국수 삼성혈점   
2480  CONT_000000000500459    관광지                       오현단   
2481  CNTS_000000000021007     숙박                 비지터게스트하우스   
2482  CNTS_200000000007258  축제/행사   한 여름밤에 즐기는 박물관 미술관 프로그램   
2483  CNTS_200000000008021     쇼핑                  에브리바디빈티지   

                        지번주소_x         위도          경도  평점   조회수  좋아요수  리뷰수  \
0       제주특별자치도 제주시 외도2동 317-1  33.494110  126.433780   0   979     0    0   
1        제주특별자치도 제주시 외도이동 1991  33.493854  126.424573   0  1043     0    0   
2       제주특별자치도 제주시 외도일동 494-5  33.488888  126.429609   0  2118     0    0   
3       제주특별자치도 제주시 외도2동 232-1  33.493509  126.436062   5  7851     0    1   
4       제주특별자치도 제주시 외도일동 125-8  33.483951  126.436436   5  5153     0    3   
...                        ...        ...         ...  ..   ...   ...  ...   
2479   제주특별자치도 제주시 이도일동 1272-4  33.506063  126.529090   5  3640     1    1   
2480   제주특별자치도 제주시 이도1동 1421-3  33.510136  126.525620   0  8418     2    0   
2481  제주특별자치도 제주시 이도1동 1349-24  33.512360  126.525215   0  1998     0    0   
2482   제주특별자치도 제주시 이도1동 1244-4  33.503182  126.534082   0  2014     1    0   
2483   제주특별자치도 제주시 이도1동 1493-1  33.510502  126.523442   0  2312     0    0   

      ...        도    읍면    시       관광객수 카드사용금액 관광지개수 유네스코유무  \
0     ...  제주특별자치도   외도동  제주시  1,389,162   19.0    25      0   
1     ...  제주특별자치도   외도동  제주시  1,389,162   19.0    25      0   
2     ...  제주특별자치도   외도동  제주시  1,389,162   19.0    25      0   
3     ...  제주특별자치도   외도동  제주시  1,389,162   19.0    25      0   
4     ...  제주특별자치도   외도동  제주시  1,389,162   19.0    25      0   
...   ...      ...   ...  ...        ...    ...   ...    ...   
2479  ...  제주특별자치도  이도1동  제주시    979,598   26.5    26      0   
2480  ...  제주특별자치도  이도1동  제주시    979,598   26.5    26      0   
2481  ...  제주특별자치도  이도1동  제주시    979,598   26.5    26      0   
2482  ...  제주특별자치도  이도1동  제주시    979,598   26.5    26      0   
2483  ...  제주특별자치도  이도1동  제주시    979,598   26.5    26      0   

                          제목_y                    지번주소_y     관광지면적  
0                         해피펜션    제주특별자치도 제주시 외도2동 317-1    2546.0  
1                       하늘타리펜션     제주특별자치도 제주시 외도이동 1991   21305.0  
2                         사방팔방    제주특별자치도 제주시 외도일동 494-5    1064.3  
3                          월대천    제주특별자치도 제주시 외도2동 232-1  123440.0  
4                      돈사촌 외도점    제주특별자치도 제주시 외도일동 125-8    6452.0  
...                        ...                       ...       ...  
2479                 만세국수 삼성혈점   제주특별자치도 제주시 이도일동 1272-4    1064.3  
2480                       오현단   제주특별자치도 제주시 이도1동 1421-3    6238.0  
2481                 비지터게스트하우스  제주특별자치도 제주시 이도1동 1349-24  262215.0  
2482   한 여름밤에 즐기는 박물관 미술관 프로그램   제주특별자치도 제주시 이도1동 1244-4     357.0  
2483                  에브리바디빈티지   제주특별자치도 제주시 이도1동 1493-1     749.0  

[2484 rows x 24 columns]

In [ ]:
ratio = pd.read_csv('/content/B_ratio_data_v4_final.csv', encoding='cp949')
ratio

지역이름 지역이름.1  켭치는 영역 픽셀 수   모든 영역 픽셀 수   ratio
0     map-7gGg    구좌읍         97078      1356567   7.16
1       anduck    조천읍        102602      1078137   9.52
2          ara    한림읍         66450       662529  10.03
3      bonggae    한경면         72620       559125  12.99
4      cheonji    대정읍         53046       549426   9.65
5        chuja    안덕면         46806       741471   6.31
6     daecheon    남원읍        134909      1340817  10.06
7      daejung    표선면         76552       956304   8.00
8       daerun    성산읍         45741       788340   5.80
9         dodu    애월읍        124784      1453041   8.59
10    donghung    예래동         24338       276144   8.81
11        ewol    중문동         52153       446328  11.68
12        guja    대천동         35949       358797  10.02
13       gunip    영천동         27173       326979   8.31
14    hangyung    효돈동         13898        45567  30.50
15      hanlim    송산동          8713        31053  28.06
16     hwabuck    대륜동         29299       161139  18.18
17      hyodon    서흥동         22615        95919  23.58
18       ido11    천지동          6807        12996  52.38
19        ido2    정방동          1803         4017  44.88
20         iho    중앙동          2315         2505  92.42
21      ildo11    동흥동         17443       103473  16.86
22       ildo2    외도동         17567        63876  27.50
23     jocheon    이호동          8760        19515  44.89
24     jungang    도두동          5908        28332  20.85
25    jungbang   용담2동         15109        36783  41.08
26     jungmun    노형동         53298       328212  16.24
27      namwon     연동         25161        90825  27.70
28     nohyung    오라동         28138       205074  13.72
29         ora    아라동         50250       514215   9.77
30      pyosun    봉개동         23031       342663   6.72
31      samdo1    삼양동         29676        74463  39.85
32      samdo2    화북동         31292        61908  50.55
33     samyang   이도2동         32473        41112  78.99
34     seohung   용담1동          3901         4845  80.52
35     songsan   삼도1동          6485         6528  99.34
36     sungsan   삼도2동          4406         6474  68.06
37         udo   일도2동         16134        17343  93.03
38       weodo   일도1동          1576         2355  66.92
39    yeahyeah    건입동          6322        21633  29.22
40        yeon   이도1동          5432         5676  95.70
41    yongdam2    우도면          1143        50685   2.26
42  youngcheon    추자면          1434        65796   2.18

In [ ]:
ratio = ratio[['지역이름.1', 'ratio']]

In [ ]:
ratio.rename(columns={'지역이름.1':"읍면"}, inplace=True)

<ipython-input-11-10fa63c46401>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ratio.rename(columns={'지역이름.1':"읍면"}, inplace=True)


In [ ]:
ratio

읍면  ratio
0    구좌읍   7.16
1    조천읍   9.52
2    한림읍  10.03
3    한경면  12.99
4    대정읍   9.65
5    안덕면   6.31
6    남원읍  10.06
7    표선면   8.00
8    성산읍   5.80
9    애월읍   8.59
10   예래동   8.81
11   중문동  11.68
12   대천동  10.02
13   영천동   8.31
14   효돈동  30.50
15   송산동  28.06
16   대륜동  18.18
17   서흥동  23.58
18   천지동  52.38
19   정방동  44.88
20   중앙동  92.42
21   동흥동  16.86
22   외도동  27.50
23   이호동  44.89
24   도두동  20.85
25  용담2동  41.08
26   노형동  16.24
27    연동  27.70
28   오라동  13.72
29   아라동   9.77
30   봉개동   6.72
31   삼양동  39.85
32   화북동  50.55
33  이도2동  78.99
34  용담1동  80.52
35  삼도1동  99.34
36  삼도2동  68.06
37  일도2동  93.03
38  일도1동  66.92
39   건입동  29.22
40  이도1동  95.70
41   우도면   2.26
42   추자면   2.18

In [ ]:
dfdf = pd.merge(df, ratio, on='읍면', how='left')

In [ ]:
dfdf.to_csv('jeju_최종.csv',encoding='cp949', index=False)